service/controller/v12/resource/instance/create.go

```diff
@@ -667,3 +695,90 @@ func isNodeDrained(drainerConfigs []corev1alpha1.DrainerConfig, instanceName str
 
 	return false
 }
+
+func (r *Resource) handleDeploymentCompletedStatus(ctx context.Context, customObject providerv1alpha1.AzureConfig) error {
+	deploymentsClient, err := r.getDeploymentsClient(ctx)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	d, err := deploymentsClient.Get(ctx, key.ClusterID(customObject), key.VmssDeploymentName)
+	if IsDeploymentNotFound(err) {
```

Comment:
We usually try to keep the reconciliation control flow at resource's top level. Here I'd propose to just mask the error and return it and then specifically handle the error in `EnsureCreated()`.